pkg/discovery/worker/compliance/taint_toleration_processor.go

```diff
@@ -30,41 +31,36 @@ type TaintTolerationProcessor struct {
 	// Map of node uid indexed by node name
 	nodeNameToUID map[string]string
 
+	cluster *repository.ClusterSummary
+
 	// Manager for schedulable nodes
-	nodesManager *SchedulableNodeManager
+	unSchedulableNodesManager *SchedulableNodeManager
 }
 
-func NewTaintTolerationProcessor(nodeAndPodGetter NodeAndPodGetter) (*TaintTolerationProcessor, error) {
-	nodes, err := nodeAndPodGetter.GetAllNodes()
```

Comment:
I have been looking at the code and at many places I find the use of interfaces and the original structure intertwined.
I think changes like this have made the code so.
Should we keep a guideline to not go back from the abstractions already defined in the code earlier? And if we choose to do so, cleanup the relevant code properly. [I guess, I would say no action needed here for now]

Reply:
Replaced the NodeAndPodGetter, that makes redundant api calls to get nodes and pods for this parsing the taints and tolerations, with the cluster summary. Cluster summary is created after node verification and holds the discovered nodes and pods objects.

Reply:
Ack!